### ISSUE TITLE
Bad SQL Upgrade Script - rename INDEX in schema

### ISSUE THREAD

**Issue by midnight-sun** [nHydrate/nHydrate#60]
I added a FK with an index in a table within a schema and renamed the column and the resultant upgrade sql is as follows, 
```
--RENAME INDEX [Voucher].[IDX_VOUCHER_ISSUINGLOCATION]
if exists (select * from sys.indexes where name = 'IDX_VOUCHER_ISSUINGLOCATION')
exec sp_rename @objname='newTable.GetSQLSchema().Voucher.IDX_VOUCHER_ISSUINGLOCATION', @newname='IDX_VOUCHER_ISSUINGLOCATIONID', @objtype='INDEX';

--DROP CONSTRAINT FOR '[Voucher].[IssuingLocation]'
DECLARE @defaultName varchar(max)
SET @defaultName = (SELECT d.name FROM sys.columns c inner join sys.default_constraints d on c.column_id = d.parent_column_id and c.object_id = d.parent_object_id inner join sys.objects o on d.parent_object_id = o.object_id where o.name = 'Voucher' and c.name = 'IssuingLocation')
if @defaultName IS NOT NULL
exec('ALTER TABLE [fishing].[Voucher] DROP CONSTRAINT ' + @defaultName)
GO
```
The upgrade fails with the following message to the console. `Either the parameter @objname is ambiguous or the claimed @objtype (INDEX) is wrong.`

it appears the generation is outputting the generation code for the rename `newTable.GetSQLSchema()` when it should be outputting the result

**Comment by midnight-sun** [comment]
/Source/nHydrate.Core.SQLGeneration/SQLEmit.cs > Line 448 has an improper interprolated string.

**Comment by midnight-sun** [comment]
So the column previously existed with an index, I renamed it, made it not null, added a relationship. 
 This upgrade is not generated in a functional manor.  Currently it is in this order;

1. rename column
2. rename index
3. make not null
4. add the foreign_key

This fails because the index exists.  I think the index should be removed and recreated.

This one is likely more difficult so I made it a new issue.

Steve
